### PR TITLE
Various cleanups of the make script.

### PR DESCRIPTION
**Pull request by tgockel**
- Adds checks for `md5`/`md5sum` and `svgo`
- Fall back to `md5sum` on systems which do not have `md5`
- Removes use of `--lossy` from `gifsicle`, as it is not a parameter
  on the program
- Removes use of `-warn` from `pngcrush`, as it is not a parameter on
  the program

### CODE REVIEW

File: make [hunk 1]
```diff
@@ -87,17 +113,17 @@ function images () {
   cp src/favicon.ico dist/favicon.ico
 
   while IFS= read -r -d '' FILE; do
-    CACHE_FILE="$CACHE_DIR/${FILE##"src/"}-$(md5 -q "$FILE")"
+    CACHE_FILE="$CACHE_DIR/${FILE##"src/"}-$(md5_hash "$FILE")"
     if ! [ -f "$CACHE_FILE" ]; then
       case ${FILE##*.} in
         "gif")
-          gifsicle --lossy --optimize=3 "$FILE" > "$CACHE_FILE"
```
Comment (jmhobbs) [inline make]: So, I locally switched to the [giflossy](https://github.com/kornelski/giflossy) fork for that feature.  Maybe we could feature detect on this?

File: make [hunk 2]
```diff
@@ -87,17 +113,17 @@ function images () {
...
+          gifsicle --optimize=3 "$FILE" > "$CACHE_FILE"
           ;;
         "jpg")
           jpegtran -progressive -optimize -copy none -outfile "$CACHE_FILE" "$FILE"
           ;;
         "png")
-          pngcrush -reduce -s -warn "$FILE" "$CACHE_FILE"
```
Comment (jmhobbs) [inline make]: For pngcrush I have 1.8.11, which has the `-warn` flag.  This was [added in 1.8.9](https://pmt.sourceforge.io/pngcrush/ChangeLog.html) it appears, so may just be a versioning issue?

Comment (tgockel) [reply make]: It is a versioning thing...now 2.0 is out, so I'm upgraded.

Comment (jmhobbs) [reply make]: Is there a fork?  I just checked homebrew and it's still `1.8.13`, not seeing a 2.0 on the project page anywhere either.

**Comment by jmhobbs** [comment]
Gonna go ahead and merge these.  giflossy can be added back later if we want.

Thanks!

**Pull request merged**